AzureCalling/External/Authentication/AADAuthHandler.swift

```diff
@@ -53,163 +74,173 @@ class AADAuthHandler {
     }
 
     // MARK: Public API
+    func login(presentingVc: UIViewController,
+               completionHandler: @escaping (Error?) -> Void) {
+
+        // Attempt silent login with existing account, falling back to interactive.
+        Task {
+            do {
+                var result: MSALResult?
+                if let account = try await loadMsalAccount() {
+                    result = try? await loginSilently(account: account)
+                }
 
-    func loadAccountAndSilentlyLogin(from viewController: UIViewController,
-                                     completionHandler: @escaping () -> Void) {
-        loadAccount { [weak self] account in
-            guard let account = account else {
-                completionHandler()
-                return
-            }
+                if result == nil {
+                    result = try await loginInteractively(from: presentingVc)
```

Comment:
I understand why we would want `loginSilently` to be async. 
But do we want the following code to be async? 
Should we maybe use a dispatchgroup? I'd like to hear your thought process behind this.

Reply:
Could you expand why you'd want a dispatch group here? 
Using `async` calls makes the code very readable and sequential.